### DISCUSSION TITLE
How to link supabase table with foreign key to User Data, using NestJS and typeorm?

### DISCUSSION THREAD

**Discussion by Queng123** [supabase/supabase#19628] [Questions]
I just create a **Supabase DB** using **NestJS** and **TypeORM**. I have some entities like this:
```ts
@Entity()
export class UserService {

    @PrimaryGeneratedColumn()
    id: number;

    @OneToOne(type => User)
    user: User;

    @OneToOne(type => Service)
    service_id: ForeignKeyMetadata;

    @Column('text')
    token: string;

    @Column({ default: () => 'CURRENT_TIMESTAMP', type: 'timestamp' })
    created: Date;

    @Column({ type: 'timestamp', default: () => 'CURRENT_TIMESTAMP', onUpdate: 'CURRENT_TIMESTAMP' })
    updated: Date;
}
```
I'm trying to link user Field to User table, that is the table for user management in supabase.

I import this like:
```ts
import { Entity, Column, PrimaryGeneratedColumn, OneToOne } from 'typeorm';
import { ForeignKeyMetadata } from 'typeorm/metadata/ForeignKeyMetadata';
import { User } from '@supabase/supabase-js';
import { Service } from './service.entity';
```

But, User is an interface, so I can't use it in (also I'm not sure I need to import that).

In the official documentation, they llink tables like:
```
create table public.profiles (
  id uuid not null references auth.users on delete cascade,
  first_name text,
  last_name text,

  primary key (id)
);
```

alter table public.profiles enable row level security;
https://supabase.com/docs/guides/auth/managing-user-data?language=js

How can I do that, using **NestJS** and **TypeORM** ? Remember, the goal of this topic is not to link a table to another table using a foreign key. The goal is to link the table to** Auth.user** table that is directly in Supabase.

**Comment by GaryAustin1** [comment]
I don't know your environment but if you have a profiles table you can populate with a user management guide style trigger on auth.users.  Then don't actually link the profiles table back to the auth.users table so "your side" does not care about auth.users.   If you need to delete when a user is deleted from auth.users then you would also need a trigger for that.
https://supabase.com/docs/guides/auth/managing-user-data#advanced-techniques

**Reply by Queng123** [reply]
Hello, thank you for your response!

I think I expressed myself badly. I want to link my profile table to an auth.users key, in order to access for example the email or the created_at field. In the doc, you can do it in this form:
```
create table public.profiles (
  id uuid not null references auth.users on delete cascade,
  first_name text,
  last_name text,

  primary key (id)
);

alter table public.profiles enable row level security;
```
I would like to do the same thing in NestJS, thanks to TypeORM (to create my table which links auth.users in a profile field). Is it clearer?
Here is a screenshot of my DB where I create the table without NestJS (using the Web interface)
![image](https://github.com/supabase/supabase/assets/91665380/2df74ab8-d603-4e3b-a5dc-ea0b1270eec4)
The goal is to create this table using NestJS and not using the Web interface

**Reply by GaryAustin1** [reply]
The suggestion I showed was to use the trigger function and populate your profile table with the email and other data you want from auth.users when the user is created.   This avoids having to link the auth schema.
Otherwise since I don't know nestjs and why you can't just link auth.users like you show in your example you'll have to see if a NestJS user comes along.
